pallets/order-book/src/order_book.rs

```diff
@@ -213,6 +214,153 @@ impl<T: crate::Config + Sized> OrderBook<T> {
+                .try_into_balance()
+                .map_err(|_| Error::<T>::AmountCalculationFailed)?;
+
+            if let Some(limit) = depth_limit {
+                match limit {
+                    OrderAmount::Base(base_limit) => {
+                        if market_base_volume + base_volume > base_limit {
+                            let delta = self.align_amount(base_limit - market_base_volume);
+                            market_base_volume += delta;
+                            market_quote_volume += (FixedWrapper::from(*price)
+                                * FixedWrapper::from(delta))
+                            .try_into_balance()
+                            .map_err(|_| Error::<T>::AmountCalculationFailed)?;
+                            enough_liquidity = true;
+                            break;
+                        }
+                    }
+                    OrderAmount::Quote(quote_limit) => {
+                        if market_quote_volume + quote_volume > quote_limit {
+                            // delta in base asset
+                            let delta = self.align_amount(
+                                (FixedWrapper::from(quote_limit - market_quote_volume)
+                                    / FixedWrapper::from(*price))
+                                .try_into_balance()
+                                .map_err(|_| Error::<T>::AmountCalculationFailed)?,
+                            );
+                            market_base_volume += delta;
+                            market_quote_volume += (FixedWrapper::from(*price)
+                                * FixedWrapper::from(delta))
+                            .try_into_balance()
+                            .map_err(|_| Error::<T>::AmountCalculationFailed)?;
+                            enough_liquidity = true;
+                            break;
+                        }
+                    }
+                }
+            }
+
+            market_base_volume += base_volume;
+            market_quote_volume += quote_volume;
+        }
+
+        ensure!(
+            depth_limit.is_none() || enough_liquidity,
+            Error::<T>::NotEnoughLiquidity
+        );
+
+        Ok((market_base_volume, market_quote_volume))
+    }
+
+    pub fn get_side(
+        &self,
+        input_asset_id: &AssetIdOf<T>,
+        output_asset_id: &AssetIdOf<T>,
+    ) -> Result<PriceVariant, DispatchError> {
+        match self.order_book_id {
+            OrderBookId::<AssetIdOf<T>> { base, quote }
+                if base == *output_asset_id && quote == *input_asset_id =>
+            {
+                Ok(PriceVariant::Buy)
+            }
+            OrderBookId::<AssetIdOf<T>> { base, quote }
+                if base == *input_asset_id && quote == *output_asset_id =>
+            {
+                Ok(PriceVariant::Sell)
+            }
+            _ => Err(Error::<T>::InvalidAsset.into()),
+        }
```

Comment:
```suggestion
        let OrderBookId::<AssetIdOf<T>> { base, quote } = self.order_book_id;
        if base == *output_asset_id && quote == *input_asset_id {
            Ok(PriceVariant::Buy)
        } else if base == *input_asset_id && quote == *output_asset_id {
            Ok(PriceVariant::Sell)
        } else {
            Err(Error::<T>::InvalidAsset.into()),
        }
```
should be easier to read

Reply:
The current version looks readable as well

Reply:
More nesting and unnecessary repeating of pattern matching is definitely less readable